packages/plugin-senpi-rewards/src/actions/checkRewardAction.ts

```diff
@@ -0,0 +1,91 @@
+import {
+    type Action,
+    type IAgentRuntime,
+    type Memory,
+    type HandlerCallback,
+    type State,
+    type ActionExample,
+} from "@moxie-protocol/core";
+import { MoxieWalletClient } from "@moxie-protocol/moxie-agent-lib/src/wallet";
+import { formatEther } from "viem";
+import {getPrice} from "../utils/codexApis";
+import { ETH, ETH_ADDRESS, ETH_TOKEN_DECIMALS, USDC, USDC_ADDRESS, USDC_TOKEN_DECIMALS } from "../utils/constants";
+import { v4 as uuidv4 } from "uuid";
+import { ethers } from "ethers";
+import { getRewardBalance } from "../utils/balance";
+
+export const checkRewardsAction: Action = {
+    name: "CHECK_REWARDS",
+    similes: [
+        "CHECK_REWARDS",
+        "VIEW_REWARDS",
+        "SHOW_REWARDS",
+        "WALLET_REWARDS",
+        "ETH_REWARDS",
+        "BASE_REWARDS",
+    ],
+    description: "Check the rewards of your agent wallet on Senpi",
+    suppressInitialMessage: true,
+    validate: async () => true,
+    handler: async (
+        runtime: IAgentRuntime,
+        message: Memory,
+        state: State,
+        _options: { [key: string]: unknown },
+        callback: HandlerCallback
+    ) => {
+  
+        const { address } = state.agentWallet as MoxieWalletClient;
+
+        console.log("address", address);
+
+        const balance = await getRewardBalance(address as `0x${string}`);
+   
+        if (!balance || balance === 0) {
+            await callback?.({
+                text: "You have no rewards to claim.",
+                action: "CHECK_REWARDS",
+            });
+            return;
+        }
+
+        const balanceAsEther = formatEther(BigInt(balance.toString()));
+
+        const rewardsInUSDCWei = await getPrice(
+            uuidv4(),
```

Comment:
Don't generate random uuid for this, instead on the top define:
```js
const traceId = message.id;
```
Then reference it here + other loggers